components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt

```diff
@@ -129,17 +150,22 @@ class CrashReporter(
 
         logger.info("Received crash: $crash")
 
-        if (CrashPrompt.shouldPromptForCrash(shouldPrompt, crash)) {
-            if (shouldSendIntent(crash)) {
-                // This crash was not fatal and the app has registered a pending intent: Send Intent to app and let the
-                // app handle showing a confirmation UI.
-                launchCrashIntent(context, crash)
+        /* Telemetry services does not require user prompts */
+        submitTelemetryCrashReport(crash)
```

Comment:
I am wondering here whether we may be executing this on the crashing process here (e.g. uncaught exception case) and whether that is problematic (after the crash it is not guaranteed that anything is working). But since we are also submitting here for the non-prompt case ... maybe that is fine? But then again we never run a non-prompt configuration in an app we ship right now.
If we step back from what we have right now... then I assume we should always launch our service here and the service will take care of uploading (non prompt case) and/or invoking telemetry from a new process? 🤔

Reply:
Agreed.  I'll simplify this so even non-prompt cases go through the service.  This way there is less that can go wrong.